derivation/derivation.go

```diff
@@ -0,0 +1,63 @@
+// nolint: govet
+package derivation
+
+import (
+	"io/ioutil"
+	"log"
+	"os"
+)
+
+type Derivation struct {
```

Comment:
Please use the `parser:""` tag format, as described in https://github.com/alecthomas/participle#tag-syntax. This will allow co-existing with other tags - let's say if we want a JSON serialization and custom field names for it.

We can probably even add the `json:""` ones, to better explain how this maps to the fields showed in the `nix show-derivation` command.